spek-junit-platform-engine/src/test/kotlin/org/jetbrains/spek/engine/LifecycleManagerTest.kt

```diff
@@ -0,0 +1,33 @@
+package org.jetbrains.spek.engine
+
+import org.jetbrains.spek.api.lifecycle.LifecycleListener
+import org.jetbrains.spek.engine.lifecycle.LifecycleManager
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import kotlin.properties.Delegates
+
+/**
+ * @author Ranie Jade Ramiso
+ */
+class LifecycleManagerTest {
+    var lifecycleManager by Delegates.notNull<LifecycleManager>()
+
+    @BeforeEach
+    fun setup() {
+        lifecycleManager = LifecycleManager()
+    }
+
+    @Test
+    fun sameListenerRegistration() {
+        val listener = object: LifecycleListener { }
+        lifecycleManager.addListener(listener)
+
+        try {
+            lifecycleManager.addListener(listener)
+            Assertions.fail("it should throw IllegalArgumentException")
+        } catch (e: IllegalArgumentException) {
+            // pass
```

Comment:
Sure, feel free to merge then! Or should I do it? Usually in projects I work on anyone can merge PR if it's approved & CI passed (we even do it automatically at work)